concept/type_/annotation.rs

```diff
@@ -168,3 +300,16 @@ impl<'a> TypeEdgePropertyEncoding<'a> for AnnotationCardinality {
         Some(Bytes::copy(bincode::serialize(&self).unwrap().as_slice()))
     }
 }
+
+impl<'a> TypeEdgePropertyEncoding<'a> for AnnotationRegex {
+    const INFIX: Infix = Infix::PropertyAnnotationRegex;
+    fn from_value_bytes<'b>(value: ByteReference<'b>) -> Self {
+        // TODO this .unwrap() should be handled as an error
```

Comment:
I do feel like we have a kind of misunderstanding of when we can use `unwrap()` and where return accurate errors (based on talks with Dmitrii and Krishnan), but I don't feel like we have time to address it right now. Just keep it in mind for the future refactoring